apis/keda/v1alpha1/scaledobject_types.go

```diff
@@ -254,21 +254,39 @@ func (so *ScaledObject) GetHPAMaxReplicas() int32 {
 	return defaultHPAMaxReplicas
 }
 
+// GetDefaultHPAMinReplicas returns defaultHPAMinReplicas
+func GetDefaultHPAMinReplicas() *int32 {
+	tmp := defaultHPAMinReplicas
+	return &tmp
+}
+
+// GetDefaultHPAMaxReplicas returns defaultHPAMaxReplicas
+func GetDefaultHPAMaxReplicas() int32 {
+	return defaultHPAMaxReplicas
+}
+
 // checkReplicaCountBoundsAreValid checks that Idle/Min/Max ReplicaCount defined in ScaledObject are correctly specified
 // i.e. that Min is not greater than Max or Idle greater or equal to Min
 func CheckReplicaCountBoundsAreValid(scaledObject *ScaledObject) error {
-	min := int32(0)
-	if scaledObject.Spec.MinReplicaCount != nil {
-		min = *scaledObject.GetHPAMinReplicas()
+	var idleReplicas *int32
+	minReplicas := *scaledObject.GetHPAMinReplicas()
+	maxReplicas := scaledObject.GetHPAMaxReplicas()
+	idleReplicas = scaledObject.Spec.IdleReplicaCount
+
+	if scaledObject.Spec.IdleReplicaCount != nil && *idleReplicas < 0 {
+		return fmt.Errorf("IdleReplicaCount=%d must not be negative", *idleReplicas)
+	}
+
+	if minReplicas < 0 {
+		return fmt.Errorf("MinReplicaCount=%d must not be negative", minReplicas)
 	}
-	max := scaledObject.GetHPAMaxReplicas()
 
-	if min > max {
-		return fmt.Errorf("MinReplicaCount=%d must be less than MaxReplicaCount=%d", min, max)
+	if minReplicas > maxReplicas {
+		return fmt.Errorf("MinReplicaCount=%d must not be greater than MaxReplicaCount=%d", minReplicas, maxReplicas)
 	}
 
-	if scaledObject.Spec.IdleReplicaCount != nil && *scaledObject.Spec.IdleReplicaCount >= min {
-		return fmt.Errorf("IdleReplicaCount=%d must be less than MinReplicaCount=%d", *scaledObject.Spec.IdleReplicaCount, min)
+	if scaledObject.Spec.IdleReplicaCount != nil && *idleReplicas >= minReplicas {
+		return fmt.Errorf("IdleReplicaCount=%d must be less than MinReplicaCount=%d", *scaledObject.Spec.IdleReplicaCount, minReplicas)
```

Comment:
Actually `idleReplicas` only can be 0 nowadays, other value will be wrong

Reply:
> Actually `idleReplicas` only can be 0 nowadays, other value will be wrong

Ok, get it.

Reply:
Here is the explanation https://keda.sh/docs/2.16/reference/scaledobject-spec/#idlereplicacount